core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -386,13 +386,15 @@ class DAGScheduler(
     def visit(rdd: RDD[_]) {
       if (!visited(rdd)) {
         visited += rdd
-        if (getCacheLocs(rdd).contains(Nil)) {
```

Comment:
As a general aside, I find `getCacheLocs(rdd).contains(Nil)` to be hard to understand to begin with.  I think that this condition is meant to be read as "if at least one partition of this RDD is not cached anywhere...".  Maybe this code would be easier to review / parse if we extracted this condition into a variable, perhaps a lazy val if we want to short-circuit, named `rddHasUncachedPartitions`, or `!rddIsCached`.